### PR TITLE
JMS :: set JMSType header

### PR DESCRIPTION
**Pull request by frederic-gendebien**
adds _type field in JmsTextMessage to be able to set the JMSType header

### CODE REVIEW

File: jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala [hunk 1]
```diff
@@ -18,25 +18,37 @@ sealed trait Destination
...
+  /**
+   * Java API: adds JMSProperty [[JmsTextMessage]]
+   */
+  def withProperty(name: String, value: Any) = copy(properties = properties + (name -> value))
+
+  /**
+   * Java API: add [[JmsTextMessage]]
+   */
+  @deprecated("Unclear method name, use withProperty instead")
+  def add(name: String, value: Any) = withProperty(name, value)
```
Comment (johanandren) [inline jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: 👍

File: jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala [hunk 2]
```diff
@@ -18,25 +18,37 @@ sealed trait Destination
 final case class Topic(name: String) extends Destination
 final case class Queue(name: String) extends Destination
 
-final case class JmsTextMessage(body: String, properties: Map[String, Any] = Map.empty) {
+final case class JmsTextMessage(body: String, _type: Option[String] = None, properties: Map[String, Any] = Map.empty) {
```
Comment (johanandren) [inline jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: Make it `type` or `jmsType` instead. Underscore is used for private fields and this is a public field.

**Comment by johanandren** [comment]
On second thought, what about adding a `JmsHeader` class and have a `Set[JmsHeader]` to make it easy to add other headers as well?

File: jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala (hunk 2 above)
Comment (frederic-gendebien) [reply jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: I agree, but I can't call it type (type is reserved) and jmsType is redundant with the name of the class, don't you think? Let's say I'll call it that way

Comment (frederic-gendebien) [reply jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: I will also add some test, right!

**Comment by frederic-gendebien** [comment]
Could be interesting, I gonna try that

**Comment by johanandren** [comment]
One way would be to make it a `sealed trait JmsHeader` and then `foreach { case JmsType(t) => textMessage.setJMSType(t) }`

**Comment by frederic-gendebien** [comment]
Funny, I already did it that way

**Comment by johanandren** [comment]
Ah, travis failed on formatting, make sure to compile both sources and tests to get the code autoformatted and check in the result of that.

**Comment by johanandren** [comment]
Fixes #522 520

**Comment by frederic-gendebien** [comment]
It is good now

**Comment by frederic-gendebien** [comment]
Thanks! Any idea when it will be released and available?

**Comment by johanandren** [comment]
We just missed the release train with a few hours, we tend to release Alpakka fairly often but we don't have a planned date for the next one

**Pull request merged**